src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -183,6 +183,69 @@ string CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::get_memory_space(const string &buf)
     return "__address_space_" + print_name(buf);
 }
 
+void CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::visit(const Call *op) {
+    if (op->call_type != Call::Intrinsic) {
+        CodeGen_C::visit(op);
+        return;
+    }
+    if (op->name == Call::interleave_vectors) {
+        int op_width = op->type.width;
+        internal_assert(op->args.size() > 0);
+        int arg_width = op->args[0].type().width;
+        if (op->args.size() == 1) {
+            // 1 argument, just do a simple assignment
+            internal_assert(op_width == arg_width);
+            print_assignment(op->type, print_expr(op->args[0]));
+        } else if (op->args.size() == 2) {
+            // 2 arguments, set the .even to the first arg and the
+            // .odd to the second arg
+            internal_assert(op->args[1].type().width == arg_width);
+            internal_assert(op_width / 2 == arg_width);
+            string a1 = print_expr(op->args[0]);
+            string a2 = print_expr(op->args[1]);
+            id = unique_name('_');
+            do_indent();
+            stream << print_type(op->type) << " " << id << ";\n";
+            do_indent();
+            stream << id << ".even = " << a1 << ";\n";
+            do_indent();
+            stream << id << ".odd = " << a2 << ";\n";
+        } else {
+            // 3+ arguments, interleave via a vector literal
+            // selecting the appropriate elements of the args
+            int dest_width = op->type.width;
+            internal_assert(dest_width <= 16);
+            int num_args = op->args.size();
+            vector<string> arg_exprs(num_args);
+            for (int i = 0; i < num_args; i++) {
+                internal_assert(op->args[i].type().width == arg_width);
+                arg_exprs[i] = print_expr(op->args[i]);
+            }
+            internal_assert(num_args * arg_width >= dest_width);
+            id = unique_name('_');
+            do_indent();
+            stream << print_type(op->type) << " " << id << ".s";
```

Comment:
This is supposed to create a new vector that rotates through the argument vectors taking the first, second, etc. from each, for example generating the following:

    int2 _1 = ...;
    int2 _2 = ...;
    int2 _3 = ...;
    int2 _4 = ...;
    int8 _5 = (int8)(_1.s0, _2.s0, _3.s0, _4.s0, _1.s1, _2.s1, _3.s1, _4.s1);

It was originally incorrect, the variable name on the left hand side of the interleave vector assignment would get suffixed with .s1234..., which is invalid syntax, my latest commit fixes this.

Also, I noticed that none of the built in correctness tests caught this when running with HL_JIT_TARGET=host-opencl. A test case should be added (probably to the interleave test) that tests interleaving of 3+ vectors.